### ISSUE TITLE
Brave tab sync keeps .lock

### ISSUE THREAD

**Issue by andbenn** [floccusaddon/floccus#1607]
### Which version of floccus are you using?

5.1.1

### How many bookmarks do you have, roughly?

12 tabs

### Are you using other means to sync bookmarks in parallel to floccus?

no

### Sync method

WebDAV

### Which browser are you using? In case you are using the phone App, specify the Android or iOS version and device please.

fastmail

### What kind of WebDAV server are you using? (if relevant)

Fastmail's webdav service

### Describe the Bug

Brave has a unique profile that only it uses to sync tabs to Fastmail's webdav service.
Since floccus was updated to 5.1.1 2-3 days ago, I've had two persistent .lock files appear.  Each time I've renamed both files then pushed my current tabs back to the server.

My Firefox also has floccus 5.1.1, but does not appear to have .lock files hanging around like Brave does.

I wonder if I should also reset the cache on the floccus tabs setting page when these lock files won't go away.

3 persistent tag profile lock files on Brave in a day.

### Expected Behavior

No lock files longer than a few minutes.  :)

I have another computer with Brave and floccus 5.1.1.  Currently running.  Will see if locks appear in that tab parofile.

### To Reproduce

Current Brave, 1.65.132
Floccus 5.1.1
Tab profile, fastmail webdav service
Tab profile is unique to each computer of mine.  No cross computer tab sharing (but bookmarks)

### Debug log provided

- [ ] I have provided a debug log file

**Comment by andbenn** [comment]
Meh,  As I leave my desktop on, woke up to a Brave tab profile lock again.  That's three.  Something's amiss.  Should I wipe out the remote files and clear the foccus cache?  Or other advice?  If you want a log, I'll run one.

**Comment by marcelklehr** [comment]
I'll try to reproduce later today and see if I can fix something

**Comment by andbenn** [comment]
No rush from me.  Prioritize as you need.  You probably have a ton of work going on. I can hold for a while.  Thank you!

**Comment by marcelklehr** [comment]
Can't reproduce :/ Can you send me some logs from this?

**Comment by marcelklehr** [comment]
Since v5.1.1 floccus should not reset the lock after it has been removed :/

**Comment by andbenn** [comment]
I've got another issue with this browser at the moment.  Floccus extension is 5.1.2.  I have a bookmark and tab profiles to sync every 15 minutes.  The check is present in the extension bar.  The profiles are "(check) All Good", but reported the "last synchronization 2 days ago".

If I change the interval, in hopes that such would trigger a sync, nothing happens.  This browser needs a reset.

**Comment by marcelklehr** [comment]
Try updating floccus first, v5.1.2 had a faulty build that wouldn't sync anymore.

**Comment by andbenn** [comment]
I might have had the bad 5.1.2.  Removed 5.1.2 from my browsers, re-installed from the extension stores, and imported my backup profiles.

I also turned on error reporting, and auto-sync.  Since doing that an hour ago, the pesky Brave browser now has it's tabs profile Scheduled and I see a .lock in my webdav storage for that profile.  Hopefully you'll get some anonymized error reporting.  Thanks!

**Comment by marcelklehr** [comment]
After two hours it should start syncing again.

**Comment by andbenn** [comment]
Additional bizarreness.  Now the two Braves on their unique tab profiles are scheduled and lock pending.

The profile files times hasn't changed since my last post.  Nor the lock files.  Seems like it forced a sync, then got into it's loop due to the lock file.  is it supposed to do that?  i thought the lock file would get removed after 2 hours.

**Comment by marcelklehr** [comment]
I would need a log from a sync run that didn't remove the lock.

**Comment by marcelklehr** [comment]
The common theme seems to be that it fails to remove the lock files on brave

**Comment by andbenn** [comment]
You may be correct.  The other Brave I have is my work laptop, a Mac.  It's Autosync'd for every 10 minutes.  I'm constantly opening/closing tabs all day.  Both it's tab profile and bookmark profile are green, up to date.  But it hasn't sync'd in 4 days.

Clicking the sync button does nothing.  I'll review it tomorrow due to the current late hour.  Thank you!

**Comment by marcelklehr** [comment]
> Clicking the sync button does nothing. 

You may need to force an update there, because of the initially faulty build in v5.1.2

**Comment by marcelklehr** [comment]
Also, do you have backups enabled for the current user in the bookmarks settings?

**Comment by G-Huber** [comment]
I am experiencing persisting lock files with Firefox on Manjaro Linux. Server side is lighttpd providing webdav.

I tried to get a log that shows this, but wasn't yet able to identify an instance of this in the logs.

Got a few thoughts on the issue though:

Unlike bookmarks tabs are fast changing. Opening and closing 3-5 tabs within one minute is absolutely possible. Maybe sync attempts are interfering with each other? 
Also unlike bookmarks closing a tab and then closing the browser is a rather usual thing to do. Maybe sync attempts don't complete? (This can't be the only reason for this behaviour though - i've seen persisting lock files even when all browsers were opened.)

I've also taken a look at the lockfiles themselves and would suggest an improvement:
Put a client (or profile) identifier in there. Currently even the client that originally created that lock and therefore holds it can't release it if the mechanism got stuck.
With an identifier attached that client would be able to know that the lock is held by itself and it's okay to proceed with their operation and finally release the lock.
This way there is at least a chance for syncing to get unstuck.

**Comment by marcelklehr** [comment]
> Put a client (or profile) identifier in there.

Yeah, I had the same idea :)

**Comment by G-Huber** [comment]
Additional strangeness: Logs (gotten via Floccus log download functionality) only ever contain a single line:

Log from 2 minutes ago:

`2024-05-18T11:16:04.483Z Resource is locked, trying again soon`

Log from 1 minute ago:

`2024-05-18T11:17:05.428Z Resource is locked, trying again soon`

Thats it. Those are the complete logs. Overwritten each time, it seems.

**Comment by marcelklehr** [comment]
Mh, if the logs are unreliable then you can debug via the inspector console:

* Go to `chrome://extensions`
* enable `Developer mode`
* click on `Service worker` button in floccus' entry next to "Inspect views: " to open the console

**Comment by G-Huber** [comment]
Even more strangeness:

The last modified date on the lock kept changing. (I made sure only one of my clients is currently up and running. Everywhere else I closed FF.)

lighttpd access.logs tell me that the client actually DOES permanently recreate that lock.

`[18/May/2024:11:20:06 +0000] mydomain 93.221.106.176 myusername "GET /webdav/browsersync/tabs_notebook.xbel.lock HTTP/2.0" 200 42 "-" "Mozilla/5.0 (X11; Linux x86_64; rv:125.0) Gecko/20100101 Firefox/125.0"
[18/May/2024:11:20:06 +0000] mydomain 93.221.106.176 myusername "GET /webdav/browsersync/tabs_notebook.xbel.lock HTTP/2.0" 200 42 "-" "Mozilla/5.0 (X11; Linux x86_64; rv:125.0) Gecko/20100101 Firefox/125.0"
[18/May/2024:11:21:06 +0000] mydomain 93.221.106.176 myusername "GET /webdav/browsersync/tabs_notebook.xbel.lock HTTP/2.0" 200 42 "-" "Mozilla/5.0 (X11; Linux x86_64; rv:125.0) Gecko/20100101 Firefox/125.0"
[18/May/2024:11:21:06 +0000] mydomain 93.221.106.176 myusername "GET /webdav/browsersync/tabs_htpc.xbel.lock HTTP/2.0" 404 341 "-" "Mozilla/5.0 (X11; Linux x86_64; rv:125.0) Gecko/20100101 Firefox/125.0"
[18/May/2024:11:21:06 +0000] mydomain 93.221.106.176 myusername "GET /webdav/browsersync/tabs_notebook.xbel.lock HTTP/2.0" 200 42 "-" "Mozilla/5.0 (X11; Linux x86_64; rv:125.0) Gecko/20100101 Firefox/125.0"
[18/May/2024:11:21:06 +0000] mydomain 93.221.106.176 myusername "PUT /webdav/browsersync/tabs_htpc.xbel.lock HTTP/2.0" 201 0 "-" "Mozilla/5.0 (X11; Linux x86_64; rv:125.0) Gecko/20100101 Firefox/125.0"
[18/May/2024:11:21:06 +0000] mydomain 93.221.106.176 myusername "GET /webdav/browsersync/tabs_htpc.xbel HTTP/2.0" 200 3395 "-" "Mozilla/5.0 (X11; Linux x86_64; rv:125.0) Gecko/20100101 Firefox/125.0"
[18/May/2024:11:21:06 +0000] mydomain 93.221.106.176 myusername "DELETE /webdav/browsersync/tabs_htpc.xbel.lock HTTP/2.0" 204 0 "-" "Mozilla/5.0 (X11; Linux x86_64; rv:125.0) Gecko/20100101 Firefox/125.0"
[18/May/2024:11:21:08 +0000] mydomain 93.221.106.176 myusername "PUT /webdav/browsersync/tabs_notebook.xbel.lock HTTP/2.0" 204 0 "-" "Mozilla/5.0 (X11; Linux x86_64; rv:125.0) Gecko/20100101 Firefox/125.0"`

This keeps going. There are constant PUT requests on that lock.

**Comment by marcelklehr** [comment]
Aha! That's like the interval timer that re-sets the lock during sync, but it should be cleared upon ending the sync run.

**Comment by marcelklehr** [comment]
I think I can reproduce this when syncing using "pull down"

**Comment by marcelklehr** [comment]
I was hoping v5.1.3 would fix this but it doesn't seem to have done that. Can you confirm?

**Comment by marcelklehr** [comment]
@G-Huber Can you try to debug via the inspector console (see above)

**Comment by G-Huber** [comment]
> I was hoping v5.1.3 would fix this but it doesn't seem to have done that. Can you confirm?
I updated two of my clients this morning and let them run AFK all day with a 15 minute sync interval on tabs. At first it looked promising - yesterday I basically got this issue immediately.
Upon returning I found a stuck lock - which somehow got unstuck just now:
All I did was start the Android client on my phone - which 

Server logs:
```
[19/May/2024:17:37:03 +0000] <mydomain> <myip> <myuser> "PUT /webdav/browsersync/tabs_htpc.xbel.lock HTTP/2.0" 204 0 "-" "Mozilla/5.0 (X11; Linux x86_64; rv:125.0) Gecko/20100101 Firefox/125.0"
[19/May/2024:17:37:03 +0000] <mydomain> <myip> <myuser> "PUT /webdav/browsersync/tabs_htpc.xbel.lock HTTP/2.0" 204 0 "-" "Mozilla/5.0 (X11; Linux x86_64; rv:125.0) Gecko/20100101 Firefox/125.0"
[19/May/2024:17:37:36 +0000] <mydomain> <myip> <myuser> "PUT /webdav/browsersync/tabs_htpc.xbel.lock HTTP/2.0" 204 0 "-" "Mozilla/5.0 (X11; Linux x86_64; rv:125.0) Gecko/20100101 Firefox/125.0"
[19/May/2024:17:37:42 +0000] <mydomain> <myip> <myuser> "GET /webdav/browsersync/tabs_htpc.xbel.lock HTTP/2.0" 200 42 "-" "Mozilla/5.0 (X11; Linux x86_64; rv:125.0) Gecko/20100101 Firefox/125.0"
[19/May/2024:17:37:42 +0000] <mydomain> <myip> <myuser> "GET /webdav/browsersync/tabs_htpc.xbel.lock HTTP/2.0" 200 42 "-" "Mozilla/5.0 (X11; Linux x86_64; rv:125.0) Gecko/20100101 Firefox/125.0"
[19/May/2024:17:38:00 +0000] <mydomain> <myip> <myuser> "GET /webdav/browsersync/tabs_htpc.xbel.lock HTTP/2.0" 200 42 "-" "Mozilla/5.0 (X11; Linux x86_64; rv:125.0) Gecko/20100101 Firefox/125.0"
[19/May/2024:17:38:45 +0000] <mydomain> <myip> <myuser> "GET /webdav/browsersync/tabs_htpc.xbel.lock HTTP/2.0" 200 42 "-" "Mozilla/5.0 (X11; Linux x86_64; rv:125.0) Gecko/20100101 Firefox/125.0"
[19/May/2024:17:38:45 +0000] <mydomain> <myip> <myuser> "GET /webdav/browsersync/tabs_htpc.xbel.lock HTTP/2.0" 200 42 "-" "Mozilla/5.0 (X11; Linux x86_64; rv:125.0) Gecko/20100101 Firefox/125.0"
[19/May/2024:17:39:00 +0000] <mydomain> <myip> <myuser> "GET /webdav/browsersync/tabs_htpc.xbel.lock HTTP/2.0" 200 42 "-" "Mozilla/5.0 (X11; Linux x86_64; rv:125.0) Gecko/20100101 Firefox/125.0"
[19/May/2024:17:39:04 +0000] <mydomain> <myip> <myuser> "PUT /webdav/browsersync/tabs_htpc.xbel.lock HTTP/2.0" 204 0 "-" "Mozilla/5.0 (X11; Linux x86_64; rv:125.0) Gecko/20100101 Firefox/125.0"
[19/May/2024:17:39:04 +0000] <mydomain> <myip> <myuser> "PUT /webdav/browsersync/tabs_htpc.xbel.lock HTTP/2.0" 204 0 "-" "Mozilla/5.0 (X11; Linux x86_64; rv:125.0) Gecko/20100101 Firefox/125.0"
[19/May/2024:17:39:37 +0000] <mydomain> <myip> <myuser> "PUT /webdav/browsersync/tabs_htpc.xbel.lock HTTP/2.0" 204 0 "-" "Mozilla/5.0 (X11; Linux x86_64; rv:125.0) Gecko/20100101 Firefox/125.0"
[19/May/2024:17:39:45 +0000] <mydomain> <myip> <myuser> "GET /webdav/browsersync/tabs_htpc.xbel.lock HTTP/2.0" 200 42 "-" "Mozilla/5.0 (X11; Linux x86_64; rv:125.0) Gecko/20100101 Firefox/125.0"
[19/May/2024:17:39:45 +0000] <mydomain> <myip> <myuser> "GET /webdav/browsersync/tabs_htpc.xbel.lock HTTP/2.0" 200 42 "-" "Mozilla/5.0 (X11; Linux x86_64; rv:125.0) Gecko/20100101 Firefox/125.0"
[19/May/2024:17:40:00 +0000] <mydomain> <myip> <myuser> "GET /webdav/browsersync/tabs_htpc.xbel.lock HTTP/2.0" 200 42 "-" "Mozilla/5.0 (X11; Linux x86_64; rv:125.0) Gecko/20100101 Firefox/125.0"
[19/May/2024:17:40:46 +0000] <mydomain> <myip> <myuser> "GET /webdav/browsersync/tabs_htpc.xbel.lock HTTP/2.0" 200 42 "-" "Mozilla/5.0 (X11; Linux x86_64; rv:125.0) Gecko/20100101 Firefox/125.0"
[19/May/2024:17:40:46 +0000] <mydomain> <myip> <myuser> "GET /webdav/browsersync/tabs_htpc.xbel.lock HTTP/2.0" 200 42 "-" "Mozilla/5.0 (X11; Linux x86_64; rv:125.0) Gecko/20100101 Firefox/125.0"
[19/May/2024:17:41:01 +0000] <mydomain> <myip> <myuser> "GET /webdav/browsersync/tabs_htpc.xbel.lock HTTP/2.0" 200 42 "-" "Mozilla/5.0 (X11; Linux x86_64; rv:125.0) Gecko/20100101 Firefox/125.0"
[19/May/2024:17:41:04 +0000] <mydomain> <myip> <myuser> "PUT /webdav/browsersync/tabs_htpc.xbel.lock HTTP/2.0" 204 0 "-" "Mozilla/5.0 (X11; Linux x86_64; rv:125.0) Gecko/20100101 Firefox/125.0"
[19/May/2024:17:41:04 +0000] <mydomain> <myip> <myuser> "PUT /webdav/browsersync/tabs_htpc.xbel.lock HTTP/2.0" 204 0 "-" "Mozilla/5.0 (X11; Linux x86_64; rv:125.0) Gecko/20100101 Firefox/125.0"
[19/May/2024:17:41:37 +0000] <mydomain> <myip> <myuser> "PUT /webdav/browsersync/tabs_htpc.xbel.lock HTTP/2.0" 204 0 "-" "Mozilla/5.0 (X11; Linux x86_64; rv:125.0) Gecko/20100101 Firefox/125.0"
[19/May/2024:17:41:47 +0000] <mydomain> <myip> <myuser> "GET /webdav/browsersync/tabs_htpc.xbel.lock HTTP/2.0" 200 42 "-" "Mozilla/5.0 (X11; Linux x86_64; rv:125.0) Gecko/20100101 Firefox/125.0"
[19/May/2024:17:41:47 +0000] <mydomain> <myip> <myuser> "GET /webdav/browsersync/tabs_htpc.xbel.lock HTTP/2.0" 200 42 "-" "Mozilla/5.0 (X11; Linux x86_64; rv:125.0) Gecko/20100101 Firefox/125.0"
[19/May/2024:17:42:01 +0000] <mydomain> <myip> <myuser> "GET /webdav/browsersync/tabs_htpc.xbel.lock HTTP/2.0" 200 42 "-" "Mozilla/5.0 (X11; Linux x86_64; rv:125.0) Gecko/20100101 Firefox/125.0"
[19/May/2024:17:42:10 +0000] <mydomain> <mysmartphoneip> <myuser> "GET /webdav/browsersync/tabs_htpc.xbel.lock HTTP/1.1" 200 42 "-" "Dalvik/2.1.0 (Linux; U; Android 13; SM-G970F Build/TQ3A.230901.001)"
[19/May/2024:17:42:10 +0000] <mydomain> <mysmartphoneip> <myuser> "GET /webdav/browsersync/tabs_notebook.xbel.lock HTTP/1.1" 404 341 "-" "Dalvik/2.1.0 (Linux; U; Android 13; SM-G970F Build/TQ3A.230901.001)"
[19/May/2024:17:42:10 +0000] <mydomain> <mysmartphoneip> <myuser> "GET /webdav/browsersync/tabs_htpc.xbel HTTP/1.1" 200 2840 "-" "Dalvik/2.1.0 (Linux; U; Android 13; SM-G970F Build/TQ3A.230901.001)"
[19/May/2024:17:42:10 +0000] <mydomain> <mysmartphoneip> <myuser> "PUT /webdav/browsersync/tabs_notebook.xbel.lock HTTP/1.1" 201 0 "-" "Dalvik/2.1.0 (Linux; U; Android 13; SM-G970F Build/TQ3A.230901.001)"
[19/May/2024:17:42:10 +0000] <mydomain> <mysmartphoneip> <myuser> "GET /webdav/browsersync/tabs_notebook.xbel HTTP/1.1" 200 5548 "-" "Dalvik/2.1.0 (Linux; U; Android 13; SM-G970F Build/TQ3A.230901.001)"
[19/May/2024:17:42:10 +0000] <mydomain> <mysmartphoneip> <myuser> "DELETE /webdav/browsersync/tabs_htpc.xbel.lock HTTP/1.1" 204 0 "-" "Dalvik/2.1.0 (Linux; U; Android 13; SM-G970F Build/TQ3A.230901.001)"
[19/May/2024:17:42:10 +0000] <mydomain> <mysmartphoneip> <myuser> "DELETE /webdav/browsersync/tabs_notebook.xbel.lock HTTP/1.1" 204 0 "-" "Dalvik/2.1.0 (Linux; U; Android 13; SM-G970F Build/TQ3A.230901.001)"
[19/May/2024:17:42:18 +0000] <mydomain> <mysmartphoneip> <myuser> "GET /webdav/browsersync/tabs_notebook.xbel.lock HTTP/1.1" 404 341 "-" "Dalvik/2.1.0 (Linux; U; Android 13; SM-G970F Build/TQ3A.230901.001)"
[19/May/2024:17:42:18 +0000] <mydomain> <mysmartphoneip> <myuser> "PUT /webdav/browsersync/tabs_notebook.xbel.lock HTTP/1.1" 201 0 "-" "Dalvik/2.1.0 (Linux; U; Android 13; SM-G970F Build/TQ3A.230901.001)"
[19/May/2024:17:42:18 +0000] <mydomain> <mysmartphoneip> <myuser> "GET /webdav/browsersync/tabs_notebook.xbel HTTP/1.1" 200 5548 "-" "Dalvik/2.1.0 (Linux; U; Android 13; SM-G970F Build/TQ3A.230901.001)"
[19/May/2024:17:42:18 +0000] <mydomain> <mysmartphoneip> <myuser> "DELETE /webdav/browsersync/tabs_notebook.xbel.lock HTTP/1.1" 204 0 "-" "Dalvik/2.1.0 (Linux; U; Android 13; SM-G970F Build/TQ3A.230901.001)"
[19/May/2024:17:42:51 +0000] <mydomain> <myip> <myuser> "GET /webdav/browsersync/tabs_htpc.xbel.lock HTTP/2.0" 404 341 "-" "Mozilla/5.0 (X11; Linux x86_64; rv:125.0) Gecko/20100101 Firefox/125.0"
[19/May/2024:17:42:51 +0000] <mydomain> <myip> <myuser> "GET /webdav/browsersync/tabs_htpc.xbel.lock HTTP/2.0" 404 341 "-" "Mozilla/5.0 (X11; Linux x86_64; rv:125.0) Gecko/20100101 Firefox/125.0"
[19/May/2024:17:42:51 +0000] <mydomain> <myip> <myuser> "PUT /webdav/browsersync/tabs_htpc.xbel.lock HTTP/2.0" 201 0 "-" "Mozilla/5.0 (X11; Linux x86_64; rv:125.0) Gecko/20100101 Firefox/125.0"
[19/May/2024:17:42:51 +0000] <mydomain> <myip> <myuser> "PUT /webdav/browsersync/tabs_htpc.xbel.lock HTTP/2.0" 204 0 "-" "Mozilla/5.0 (X11; Linux x86_64; rv:125.0) Gecko/20100101 Firefox/125.0"
[19/May/2024:17:42:51 +0000] <mydomain> <myip> <myuser> "GET /webdav/browsersync/tabs_htpc.xbel HTTP/2.0" 200 2840 "-" "Mozilla/5.0 (X11; Linux x86_64; rv:125.0) Gecko/20100101 Firefox/125.0"
[19/May/2024:17:42:51 +0000] <mydomain> <myip> <myuser> "GET /webdav/browsersync/tabs_htpc.xbel HTTP/2.0" 200 2840 "-" "Mozilla/5.0 (X11; Linux x86_64; rv:125.0) Gecko/20100101 Firefox/125.0"
[19/May/2024:17:42:51 +0000] <mydomain> <myip> <myuser> "PUT /webdav/browsersync/tabs_htpc.xbel HTTP/2.0" 204 0 "-" "Mozilla/5.0 (X11; Linux x86_64; rv:125.0) Gecko/20100101 Firefox/125.0"
[19/May/2024:17:42:51 +0000] <mydomain> <myip> <myuser> "PUT /webdav/browsersync/tabs_htpc.xbel HTTP/2.0" 204 0 "-" "Mozilla/5.0 (X11; Linux x86_64; rv:125.0) Gecko/20100101 Firefox/125.0"
[19/May/2024:17:42:51 +0000] <mydomain> <myip> <myuser> "DELETE /webdav/browsersync/tabs_htpc.xbel.lock HTTP/2.0" 204 0 "-" "Mozilla/5.0 (X11; Linux x86_64; rv:125.0) Gecko/20100101 Firefox/125.0"
[19/May/2024:17:42:51 +0000] <mydomain> <myip> <myuser> "DELETE /webdav/browsersync/tabs_htpc.xbel.lock HTTP/2.0" 404 341 "-" "Mozilla/5.0 (X11; Linux x86_64; rv:125.0) Gecko/20100101 Firefox/125.0"
```
It looks like one of the two clients somehow tries to do everything twice...

> @G-Huber Can you try to debug via the inspector console (see above)
I'm on Firefox as I mentioned. I tried to find the equivalent there but haven't been successful...

In the extension storage there is a logs array - but with a stuck lock this only contains a single entry - same as the logs I provided above.

If you could point me to the right place I'll check once there is another stuck lock.

**Comment by marcelklehr** [comment]
> I'm on Firefox as I mentioned. I tried to find the equivalent there but haven't been successful...

Right.

#### Firefox
* Go to `about:debugging`
* Go to `This firefox`
* click on `debug` button next to floccus entry to open the console

**Comment by G-Huber** [comment]
I think I managed to catch a lock getting stuck:

Server
```
[19/May/2024:17:43:01 +0000] <mydomain> <myip> <myuser> "GET /webdav/browsersync/tabs_htpc.xbel.lock HTTP/2.0" 404 341 "-" "Mozilla/5.0 (X11; Linux x86_64; rv:125.0) Gecko/20100101 Firefox/125.0"
[19/May/2024:17:43:01 +0000] <mydomain> <myip> <myuser> "PUT /webdav/browsersync/tabs_htpc.xbel.lock HTTP/2.0" 201 0 "-" "Mozilla/5.0 (X11; Linux x86_64; rv:125.0) Gecko/20100101 Firefox/125.0"
[19/May/2024:17:43:01 +0000] <mydomain> <myip> <myuser> "GET /webdav/browsersync/tabs_htpc.xbel HTTP/2.0" 200 3210 "-" "Mozilla/5.0 (X11; Linux x86_64; rv:125.0) Gecko/20100101 Firefox/125.0"
[19/May/2024:17:43:02 +0000] <mydomain> <myip> <myuser> "DELETE /webdav/browsersync/tabs_htpc.xbel.lock HTTP/2.0" 204 0 "-" "Mozilla/5.0 (X11; Linux x86_64; rv:125.0) Gecko/20100101 Firefox/125.0"
[19/May/2024:17:43:04 +0000] <mydomain> <myip> <myuser> "PUT /webdav/browsersync/tabs_htpc.xbel.lock HTTP/2.0" 201 0 "-" "Mozilla/5.0 (X11; Linux x86_64; rv:125.0) Gecko/20100101 Firefox/125.0"
[19/May/2024:17:43:04 +0000] <mydomain> <myip> <myuser> "PUT /webdav/browsersync/tabs_htpc.xbel.lock HTTP/2.0" 204 0 "-" "Mozilla/5.0 (X11; Linux x86_64; rv:125.0) Gecko/20100101 Firefox/125.0"
[19/May/2024:17:43:37 +0000] <mydomain> <myip> <myuser> "PUT /webdav/browsersync/tabs_htpc.xbel.lock HTTP/2.0" 204 0 "-" "Mozilla/5.0 (X11; Linux x86_64; rv:125.0) Gecko/20100101 Firefox/125.0"
[19/May/2024:17:44:51 +0000] <mydomain> <myip> <myuser> "PUT /webdav/browsersync/tabs_htpc.xbel.lock HTTP/2.0" 204 0 "-" "Mozilla/5.0 (X11; Linux x86_64; rv:125.0) Gecko/20100101 Firefox/125.0"
[19/May/2024:17:45:04 +0000] <mydomain> <myip> <myuser> "PUT /webdav/browsersync/tabs_htpc.xbel.lock HTTP/2.0" 204 0 "-" "Mozilla/5.0 (X11; Linux x86_64; rv:125.0) Gecko/20100101 Firefox/125.0"
[19/May/2024:17:45:04 +0000] <mydomain> <myip> <myuser> "PUT /webdav/browsersync/tabs_htpc.xbel.lock HTTP/2.0" 204 0 "-" "Mozilla/5.0 (X11; Linux x86_64; rv:125.0) Gecko/20100101 Firefox/125.0"
[19/May/2024:17:45:37 +0000] <mydomain> <myip> <myuser> "PUT /webdav/browsersync/tabs_htpc.xbel.lock HTTP/2.0" 204 0 "-" "Mozilla/5.0 (X11; Linux x86_64; rv:125.0) Gecko/20100101 Firefox/125.0"
[19/May/2024:17:45:52 +0000] <mydomain> <myip> <myuser> "GET /webdav/browsersync/tabs_notebook.xbel.lock HTTP/2.0" 404 341 "-" "Mozilla/5.0 (X11; Linux x86_64; rv:125.0) Gecko/20100101 Firefox/125.0"
[19/May/2024:17:45:52 +0000] <mydomain> <myip> <myuser> "PUT /webdav/browsersync/tabs_notebook.xbel.lock HTTP/2.0" 201 0 "-" "Mozilla/5.0 (X11; Linux x86_64; rv:125.0) Gecko/20100101 Firefox/125.0"
[19/May/2024:17:45:52 +0000] <mydomain> <myip> <myuser> "GET /webdav/browsersync/tabs_notebook.xbel HTTP/2.0" 200 5548 "-" "Mozilla/5.0 (X11; Linux x86_64; rv:125.0) Gecko/20100101 Firefox/125.0"
[19/May/2024:17:45:52 +0000] <mydomain> <myip> <myuser> "DELETE /webdav/browsersync/tabs_notebook.xbel.lock HTTP/2.0" 204 0 "-" "Mozilla/5.0 (X11; Linux x86_64; rv:125.0) Gecko/20100101 Firefox/125.0"
[19/May/2024:17:46:53 +0000] <mydomain> <myip> <myuser> "PUT /webdav/browsersync/tabs_htpc.xbel.lock HTTP/2.0" 204 0 "-" "Mozilla/5.0 (X11; Linux x86_64; rv:125.0) Gecko/20100101 Firefox/125.0"
[19/May/2024:17:47:04 +0000] <mydomain> <myip> <myuser> "PUT /webdav/browsersync/tabs_htpc.xbel.lock HTTP/2.0" 204 0 "-" "Mozilla/5.0 (X11; Linux x86_64; rv:125.0) Gecko/20100101 Firefox/125.0"
[19/May/2024:17:47:04 +0000] <mydomain> <myip> <myuser> "PUT /webdav/browsersync/tabs_htpc.xbel.lock HTTP/2.0" 204 0 "-" "Mozilla/5.0 (X11; Linux x86_64; rv:125.0) Gecko/20100101 Firefox/125.0"
[19/May/2024:17:47:37 +0000] <mydomain> <myip> <myuser> "PUT /webdav/browsersync/tabs_htpc.xbel.lock HTTP/2.0" 204 0 "-" "Mozilla/5.0 (X11; Linux x86_64; rv:125.0) Gecko/20100101 Firefox/125.0"
[19/May/2024:17:48:53 +0000] <mydomain> <myip> <myuser> "PUT /webdav/browsersync/tabs_htpc.xbel.lock HTTP/2.0" 204 0 "-" "Mozilla/5.0 (X11; Linux x86_64; rv:125.0) Gecko/20100101 Firefox/125.0"
[19/May/2024:17:49:04 +0000] <mydomain> <myip> <myuser> "PUT /webdav/browsersync/tabs_htpc.xbel.lock HTTP/2.0" 204 0 "-" "Mozilla/5.0 (X11; Linux x86_64; rv:125.0) Gecko/20100101 Firefox/125.0"
[19/May/2024:17:49:04 +0000] <mydomain> <myip> <myuser> "PUT /webdav/browsersync/tabs_htpc.xbel.lock HTTP/2.0" 204 0 "-" "Mozilla/5.0 (X11; Linux x86_64; rv:125.0) Gecko/20100101 Firefox/125.0"
[19/May/2024:17:49:37 +0000] <mydomain> <myip> <myuser> "PUT /webdav/browsersync/tabs_htpc.xbel.lock HTTP/2.0" 204 0 "-" "Mozilla/5.0 (X11; Linux x86_64; rv:125.0) Gecko/20100101 Firefox/125.0"
[19/May/2024:17:50:53 +0000] <mydomain> <myip> <myuser> "PUT /webdav/browsersync/tabs_htpc.xbel.lock HTTP/2.0" 204 0 "-" "Mozilla/5.0 (X11; Linux x86_64; rv:125.0) Gecko/20100101 Firefox/125.0"
[19/May/2024:17:51:03 +0000] <mydomain> <myip> <myuser> "GET /webdav/browsersync/tabs_notebook.xbel.lock HTTP/2.0" 404 341 "-" "Mozilla/5.0 (X11; Linux x86_64; rv:125.0) Gecko/20100101 Firefox/125.0"
[19/May/2024:17:51:03 +0000] <mydomain> <myip> <myuser> "PUT /webdav/browsersync/tabs_notebook.xbel.lock HTTP/2.0" 201 0 "-" "Mozilla/5.0 (X11; Linux x86_64; rv:125.0) Gecko/20100101 Firefox/125.0"
[19/May/2024:17:51:03 +0000] <mydomain> <myip> <myuser> "GET /webdav/browsersync/tabs_notebook.xbel HTTP/2.0" 200 5548 "-" "Mozilla/5.0 (X11; Linux x86_64; rv:125.0) Gecko/20100101 Firefox/125.0"
[19/May/2024:17:51:03 +0000] <mydomain> <myip> <myuser> "DELETE /webdav/browsersync/tabs_notebook.xbel.lock HTTP/2.0" 204 0 "-" "Mozilla/5.0 (X11; Linux x86_64; rv:125.0) Gecko/20100101 Firefox/125.0"
[19/May/2024:17:51:04 +0000] <mydomain> <myip> <myuser> "PUT /webdav/browsersync/tabs_htpc.xbel.lock HTTP/2.0" 204 0 "-" "Mozilla/5.0 (X11; Linux x86_64; rv:125.0) Gecko/20100101 Firefox/125.0"
[19/May/2024:17:51:04 +0000] <mydomain> <myip> <myuser> "PUT /webdav/browsersync/tabs_htpc.xbel.lock HTTP/2.0" 204 0 "-" "Mozilla/5.0 (X11; Linux x86_64; rv:125.0) Gecko/20100101 Firefox/125.0"
[19/May/2024:17:51:37 +0000] <mydomain> <myip> <myuser> "PUT /webdav/browsersync/tabs_htpc.xbel.lock HTTP/2.0" 204 0 "-" "Mozilla/5.0 (X11; Linux x86_64; rv:125.0) Gecko/20100101 Firefox/125.0"
[19/May/2024:17:52:53 +0000] <mydomain> <myip> <myuser> "PUT /webdav/browsersync/tabs_htpc.xbel.lock HTTP/2.0" 204 0 "-" "Mozilla/5.0 (X11; Linux x86_64; rv:125.0) Gecko/20100101 Firefox/125.0"
[19/May/2024:17:53:04 +0000] <mydomain> <myip> <myuser> "PUT /webdav/browsersync/tabs_htpc.xbel.lock HTTP/2.0" 204 0 "-" "Mozilla/5.0 (X11; Linux x86_64; rv:125.0) Gecko/20100101 Firefox/125.0"
[19/May/2024:17:53:04 +0000] <mydomain> <myip> <myuser> "PUT /webdav/browsersync/tabs_htpc.xbel.lock HTTP/2.0" 204 0 "-" "Mozilla/5.0 (X11; Linux x86_64; rv:125.0) Gecko/20100101 Firefox/125.0"
[19/May/2024:17:53:37 +0000] <mydomain> <myip> <myuser> "PUT /webdav/browsersync/tabs_htpc.xbel.lock HTTP/2.0" 204 0 "-" "Mozilla/5.0 (X11; Linux x86_64; rv:125.0) Gecko/20100101 Firefox/125.0"
[19/May/2024:17:54:53 +0000] <mydomain> <myip> <myuser> "PUT /webdav/browsersync/tabs_htpc.xbel.lock HTTP/2.0" 204 0 "-" "Mozilla/5.0 (X11; Linux x86_64; rv:125.0) Gecko/20100101 Firefox/125.0"
[19/May/2024:17:55:04 +0000] <mydomain> <myip> <myuser> "PUT /webdav/browsersync/tabs_htpc.xbel.lock HTTP/2.0" 204 0 "-" "Mozilla/5.0 (X11; Linux x86_64; rv:125.0) Gecko/20100101 Firefox/125.0"
[19/May/2024:17:55:04 +0000] <mydomain> <myip> <myuser> "PUT /webdav/browsersync/tabs_htpc.xbel.lock HTTP/2.0" 204 0 "-" "Mozilla/5.0 (X11; Linux x86_64; rv:125.0) Gecko/20100101 Firefox/125.0"
[19/May/2024:17:55:37 +0000] <mydomain> <myip> <myuser> "PUT /webdav/browsersync/tabs_htpc.xbel.lock HTTP/2.0" 204 0 "-" "Mozilla/5.0 (X11; Linux x86_64; rv:125.0) Gecko/20100101 Firefox/125.0"
[19/May/2024:17:56:53 +0000] <mydomain> <myip> <myuser> "PUT /webdav/browsersync/tabs_htpc.xbel.lock HTTP/2.0" 204 0 "-" "Mozilla/5.0 (X11; Linux x86_64; rv:125.0) Gecko/20100101 Firefox/125.0"
[19/May/2024:17:57:04 +0000] <mydomain> <myip> <myuser> "PUT /webdav/browsersync/tabs_htpc.xbel.lock HTTP/2.0" 204 0 "-" "Mozilla/5.0 (X11; Linux x86_64; rv:125.0) Gecko/20100101 Firefox/125.0"
[19/May/2024:17:57:04 +0000] <mydomain> <myip> <myuser> "PUT /webdav/browsersync/tabs_htpc.xbel.lock HTTP/2.0" 204 0 "-" "Mozilla/5.0 (X11; Linux x86_64; rv:125.0) Gecko/20100101 Firefox/125.0"
[19/May/2024:17:57:37 +0000] <mydomain> <myip> <myuser> "PUT /webdav/browsersync/tabs_htpc.xbel.lock HTTP/2.0" 204 0 "-" "Mozilla/5.0 (X11; Linux x86_64; rv:125.0) Gecko/20100101 Firefox/125.0"
[19/May/2024:17:58:00 +0000] <mydomain> <myip> <myuser> "GET /webdav/browsersync/tabs_htpc.xbel.lock HTTP/2.0" 200 42 "-" "Mozilla/5.0 (X11; Linux x86_64; rv:125.0) Gecko/20100101 Firefox/125.0"
```

Client
```
"2024-05-19T17:46:53.259Z Setting lock: https://www.kmailer.eu/webdav/browsersync/tabs_htpc.xbel.lock",
"2024-05-19T17:47:04.032Z Setting lock: https://www.kmailer.eu/webdav/browsersync/tabs_htpc.xbel.lock",
"2024-05-19T17:47:04.036Z Setting lock: https://www.kmailer.eu/webdav/browsersync/tabs_htpc.xbel.lock",
"2024-05-19T17:47:37.249Z Setting lock: https://www.kmailer.eu/webdav/browsersync/tabs_htpc.xbel.lock",
"2024-05-19T17:48:53.259Z Setting lock: https://www.kmailer.eu/webdav/browsersync/tabs_htpc.xbel.lock",
"2024-05-19T17:49:04.032Z Setting lock: https://www.kmailer.eu/webdav/browsersync/tabs_htpc.xbel.lock",
"2024-05-19T17:49:04.035Z Setting lock: https://www.kmailer.eu/webdav/browsersync/tabs_htpc.xbel.lock",
"2024-05-19T17:49:37.248Z Setting lock: https://www.kmailer.eu/webdav/browsersync/tabs_htpc.xbel.lock",
"2024-05-19T17:50:53.260Z Setting lock: https://www.kmailer.eu/webdav/browsersync/tabs_htpc.xbel.lock",
"2024-05-19T17:51:04.032Z Setting lock: https://www.kmailer.eu/webdav/browsersync/tabs_htpc.xbel.lock",
"2024-05-19T17:51:04.037Z Setting lock: https://www.kmailer.eu/webdav/browsersync/tabs_htpc.xbel.lock",
"2024-05-19T17:51:37.248Z Setting lock: https://www.kmailer.eu/webdav/browsersync/tabs_htpc.xbel.lock",
"2024-05-19T17:52:53.261Z Setting lock: https://www.kmailer.eu/webdav/browsersync/tabs_htpc.xbel.lock",
"2024-05-19T17:53:04.033Z Setting lock: https://www.kmailer.eu/webdav/browsersync/tabs_htpc.xbel.lock",
"2024-05-19T17:53:04.037Z Setting lock: https://www.kmailer.eu/webdav/browsersync/tabs_htpc.xbel.lock",
"2024-05-19T17:53:37.250Z Setting lock: https://www.kmailer.eu/webdav/browsersync/tabs_htpc.xbel.lock",
"2024-05-19T17:54:53.261Z Setting lock: https://www.kmailer.eu/webdav/browsersync/tabs_htpc.xbel.lock",
"2024-05-19T17:55:04.033Z Setting lock: https://www.kmailer.eu/webdav/browsersync/tabs_htpc.xbel.lock",
"2024-05-19T17:55:04.038Z Setting lock: https://www.kmailer.eu/webdav/browsersync/tabs_htpc.xbel.lock",
"2024-05-19T17:55:37.250Z Setting lock: https://www.kmailer.eu/webdav/browsersync/tabs_htpc.xbel.lock",
"2024-05-19T17:56:53.261Z Setting lock: https://www.kmailer.eu/webdav/browsersync/tabs_htpc.xbel.lock",
"2024-05-19T17:57:04.034Z Setting lock: https://www.kmailer.eu/webdav/browsersync/tabs_htpc.xbel.lock",
"2024-05-19T17:57:04.038Z Setting lock: https://www.kmailer.eu/webdav/browsersync/tabs_htpc.xbel.lock",
"2024-05-19T17:57:37.255Z Setting lock: https://www.kmailer.eu/webdav/browsersync/tabs_htpc.xbel.lock",
"2024-05-19T17:58:00.094Z Starting sync process for account panda@www.kmailer.eu",
"2024-05-19T17:58:00.104Z onSyncStart: begin",
"2024-05-19T17:58:00.106Z https://www.kmailer.eu/webdav/browsersync/tabs_htpc.xbel.lock",
"2024-05-19T17:58:00.181Z Resource is locked, trying again soon"
```

Firefox addon debug console is flooded with two types of messages:
- Uncaugt (in promise) Error: Could not establish connection. Receiving end does not exist.
- TypeError: n.default.action is undefined

If I filter down to "log" nothing is left.

**Comment by G-Huber** [comment]
I think I managed to catch a lock getting stuck:

Server
```
[19/May/2024:17:43:01 +0000] <mydomain> <myip> <myuser> "GET /webdav/browsersync/tabs_htpc.xbel.lock HTTP/2.0" 404 341 "-" "Mozilla/5.0 (X11; Linux x86_64; rv:125.0) Gecko/20100101 Firefox/125.0"
[19/May/2024:17:43:01 +0000] <mydomain> <myip> <myuser> "PUT /webdav/browsersync/tabs_htpc.xbel.lock HTTP/2.0" 201 0 "-" "Mozilla/5.0 (X11; Linux x86_64; rv:125.0) Gecko/20100101 Firefox/125.0"
[19/May/2024:17:43:01 +0000] <mydomain> <myip> <myuser> "GET /webdav/browsersync/tabs_htpc.xbel HTTP/2.0" 200 3210 "-" "Mozilla/5.0 (X11; Linux x86_64; rv:125.0) Gecko/20100101 Firefox/125.0"
[19/May/2024:17:43:02 +0000] <mydomain> <myip> <myuser> "DELETE /webdav/browsersync/tabs_htpc.xbel.lock HTTP/2.0" 204 0 "-" "Mozilla/5.0 (X11; Linux x86_64; rv:125.0) Gecko/20100101 Firefox/125.0"
[19/May/2024:17:43:04 +0000] <mydomain> <myip> <myuser> "PUT /webdav/browsersync/tabs_htpc.xbel.lock HTTP/2.0" 201 0 "-" "Mozilla/5.0 (X11; Linux x86_64; rv:125.0) Gecko/20100101 Firefox/125.0"
[19/May/2024:17:43:04 +0000] <mydomain> <myip> <myuser> "PUT /webdav/browsersync/tabs_htpc.xbel.lock HTTP/2.0" 204 0 "-" "Mozilla/5.0 (X11; Linux x86_64; rv:125.0) Gecko/20100101 Firefox/125.0"
[19/May/2024:17:43:37 +0000] <mydomain> <myip> <myuser> "PUT /webdav/browsersync/tabs_htpc.xbel.lock HTTP/2.0" 204 0 "-" "Mozilla/5.0 (X11; Linux x86_64; rv:125.0) Gecko/20100101 Firefox/125.0"
[19/May/2024:17:44:51 +0000] <mydomain> <myip> <myuser> "PUT /webdav/browsersync/tabs_htpc.xbel.lock HTTP/2.0" 204 0 "-" "Mozilla/5.0 (X11; Linux x86_64; rv:125.0) Gecko/20100101 Firefox/125.0"
[19/May/2024:17:45:04 +0000] <mydomain> <myip> <myuser> "PUT /webdav/browsersync/tabs_htpc.xbel.lock HTTP/2.0" 204 0 "-" "Mozilla/5.0 (X11; Linux x86_64; rv:125.0) Gecko/20100101 Firefox/125.0"
[19/May/2024:17:45:04 +0000] <mydomain> <myip> <myuser> "PUT /webdav/browsersync/tabs_htpc.xbel.lock HTTP/2.0" 204 0 "-" "Mozilla/5.0 (X11; Linux x86_64; rv:125.0) Gecko/20100101 Firefox/125.0"
[19/May/2024:17:45:37 +0000] <mydomain> <myip> <myuser> "PUT /webdav/browsersync/tabs_htpc.xbel.lock HTTP/2.0" 204 0 "-" "Mozilla/5.0 (X11; Linux x86_64; rv:125.0) Gecko/20100101 Firefox/125.0"
[19/May/2024:17:45:52 +0000] <mydomain> <myip> <myuser> "GET /webdav/browsersync/tabs_notebook.xbel.lock HTTP/2.0" 404 341 "-" "Mozilla/5.0 (X11; Linux x86_64; rv:125.0) Gecko/20100101 Firefox/125.0"
[19/May/2024:17:45:52 +0000] <mydomain> <myip> <myuser> "PUT /webdav/browsersync/tabs_notebook.xbel.lock HTTP/2.0" 201 0 "-" "Mozilla/5.0 (X11; Linux x86_64; rv:125.0) Gecko/20100101 Firefox/125.0"
[19/May/2024:17:45:52 +0000] <mydomain> <myip> <myuser> "GET /webdav/browsersync/tabs_notebook.xbel HTTP/2.0" 200 5548 "-" "Mozilla/5.0 (X11; Linux x86_64; rv:125.0) Gecko/20100101 Firefox/125.0"
[19/May/2024:17:45:52 +0000] <mydomain> <myip> <myuser> "DELETE /webdav/browsersync/tabs_notebook.xbel.lock HTTP/2.0" 204 0 "-" "Mozilla/5.0 (X11; Linux x86_64; rv:125.0) Gecko/20100101 Firefox/125.0"
[19/May/2024:17:46:53 +0000] <mydomain> <myip> <myuser> "PUT /webdav/browsersync/tabs_htpc.xbel.lock HTTP/2.0" 204 0 "-" "Mozilla/5.0 (X11; Linux x86_64; rv:125.0) Gecko/20100101 Firefox/125.0"
[19/May/2024:17:47:04 +0000] <mydomain> <myip> <myuser> "PUT /webdav/browsersync/tabs_htpc.xbel.lock HTTP/2.0" 204 0 "-" "Mozilla/5.0 (X11; Linux x86_64; rv:125.0) Gecko/20100101 Firefox/125.0"
[19/May/2024:17:47:04 +0000] <mydomain> <myip> <myuser> "PUT /webdav/browsersync/tabs_htpc.xbel.lock HTTP/2.0" 204 0 "-" "Mozilla/5.0 (X11; Linux x86_64; rv:125.0) Gecko/20100101 Firefox/125.0"
[19/May/2024:17:47:37 +0000] <mydomain> <myip> <myuser> "PUT /webdav/browsersync/tabs_htpc.xbel.lock HTTP/2.0" 204 0 "-" "Mozilla/5.0 (X11; Linux x86_64; rv:125.0) Gecko/20100101 Firefox/125.0"
[19/May/2024:17:48:53 +0000] <mydomain> <myip> <myuser> "PUT /webdav/browsersync/tabs_htpc.xbel.lock HTTP/2.0" 204 0 "-" "Mozilla/5.0 (X11; Linux x86_64; rv:125.0) Gecko/20100101 Firefox/125.0"
[19/May/2024:17:49:04 +0000] <mydomain> <myip> <myuser> "PUT /webdav/browsersync/tabs_htpc.xbel.lock HTTP/2.0" 204 0 "-" "Mozilla/5.0 (X11; Linux x86_64; rv:125.0) Gecko/20100101 Firefox/125.0"
[19/May/2024:17:49:04 +0000] <mydomain> <myip> <myuser> "PUT /webdav/browsersync/tabs_htpc.xbel.lock HTTP/2.0" 204 0 "-" "Mozilla/5.0 (X11; Linux x86_64; rv:125.0) Gecko/20100101 Firefox/125.0"
[19/May/2024:17:49:37 +0000] <mydomain> <myip> <myuser> "PUT /webdav/browsersync/tabs_htpc.xbel.lock HTTP/2.0" 204 0 "-" "Mozilla/5.0 (X11; Linux x86_64; rv:125.0) Gecko/20100101 Firefox/125.0"
[19/May/2024:17:50:53 +0000] <mydomain> <myip> <myuser> "PUT /webdav/browsersync/tabs_htpc.xbel.lock HTTP/2.0" 204 0 "-" "Mozilla/5.0 (X11; Linux x86_64; rv:125.0) Gecko/20100101 Firefox/125.0"
[19/May/2024:17:51:03 +0000] <mydomain> <myip> <myuser> "GET /webdav/browsersync/tabs_notebook.xbel.lock HTTP/2.0" 404 341 "-" "Mozilla/5.0 (X11; Linux x86_64; rv:125.0) Gecko/20100101 Firefox/125.0"
[19/May/2024:17:51:03 +0000] <mydomain> <myip> <myuser> "PUT /webdav/browsersync/tabs_notebook.xbel.lock HTTP/2.0" 201 0 "-" "Mozilla/5.0 (X11; Linux x86_64; rv:125.0) Gecko/20100101 Firefox/125.0"
[19/May/2024:17:51:03 +0000] <mydomain> <myip> <myuser> "GET /webdav/browsersync/tabs_notebook.xbel HTTP/2.0" 200 5548 "-" "Mozilla/5.0 (X11; Linux x86_64; rv:125.0) Gecko/20100101 Firefox/125.0"
[19/May/2024:17:51:03 +0000] <mydomain> <myip> <myuser> "DELETE /webdav/browsersync/tabs_notebook.xbel.lock HTTP/2.0" 204 0 "-" "Mozilla/5.0 (X11; Linux x86_64; rv:125.0) Gecko/20100101 Firefox/125.0"
[19/May/2024:17:51:04 +0000] <mydomain> <myip> <myuser> "PUT /webdav/browsersync/tabs_htpc.xbel.lock HTTP/2.0" 204 0 "-" "Mozilla/5.0 (X11; Linux x86_64; rv:125.0) Gecko/20100101 Firefox/125.0"
[19/May/2024:17:51:04 +0000] <mydomain> <myip> <myuser> "PUT /webdav/browsersync/tabs_htpc.xbel.lock HTTP/2.0" 204 0 "-" "Mozilla/5.0 (X11; Linux x86_64; rv:125.0) Gecko/20100101 Firefox/125.0"
[19/May/2024:17:51:37 +0000] <mydomain> <myip> <myuser> "PUT /webdav/browsersync/tabs_htpc.xbel.lock HTTP/2.0" 204 0 "-" "Mozilla/5.0 (X11; Linux x86_64; rv:125.0) Gecko/20100101 Firefox/125.0"
[19/May/2024:17:52:53 +0000] <mydomain> <myip> <myuser> "PUT /webdav/browsersync/tabs_htpc.xbel.lock HTTP/2.0" 204 0 "-" "Mozilla/5.0 (X11; Linux x86_64; rv:125.0) Gecko/20100101 Firefox/125.0"
[19/May/2024:17:53:04 +0000] <mydomain> <myip> <myuser> "PUT /webdav/browsersync/tabs_htpc.xbel.lock HTTP/2.0" 204 0 "-" "Mozilla/5.0 (X11; Linux x86_64; rv:125.0) Gecko/20100101 Firefox/125.0"
[19/May/2024:17:53:04 +0000] <mydomain> <myip> <myuser> "PUT /webdav/browsersync/tabs_htpc.xbel.lock HTTP/2.0" 204 0 "-" "Mozilla/5.0 (X11; Linux x86_64; rv:125.0) Gecko/20100101 Firefox/125.0"
[19/May/2024:17:53:37 +0000] <mydomain> <myip> <myuser> "PUT /webdav/browsersync/tabs_htpc.xbel.lock HTTP/2.0" 204 0 "-" "Mozilla/5.0 (X11; Linux x86_64; rv:125.0) Gecko/20100101 Firefox/125.0"
[19/May/2024:17:54:53 +0000] <mydomain> <myip> <myuser> "PUT /webdav/browsersync/tabs_htpc.xbel.lock HTTP/2.0" 204 0 "-" "Mozilla/5.0 (X11; Linux x86_64; rv:125.0) Gecko/20100101 Firefox/125.0"
[19/May/2024:17:55:04 +0000] <mydomain> <myip> <myuser> "PUT /webdav/browsersync/tabs_htpc.xbel.lock HTTP/2.0" 204 0 "-" "Mozilla/5.0 (X11; Linux x86_64; rv:125.0) Gecko/20100101 Firefox/125.0"
[19/May/2024:17:55:04 +0000] <mydomain> <myip> <myuser> "PUT /webdav/browsersync/tabs_htpc.xbel.lock HTTP/2.0" 204 0 "-" "Mozilla/5.0 (X11; Linux x86_64; rv:125.0) Gecko/20100101 Firefox/125.0"
[19/May/2024:17:55:37 +0000] <mydomain> <myip> <myuser> "PUT /webdav/browsersync/tabs_htpc.xbel.lock HTTP/2.0" 204 0 "-" "Mozilla/5.0 (X11; Linux x86_64; rv:125.0) Gecko/20100101 Firefox/125.0"
[19/May/2024:17:56:53 +0000] <mydomain> <myip> <myuser> "PUT /webdav/browsersync/tabs_htpc.xbel.lock HTTP/2.0" 204 0 "-" "Mozilla/5.0 (X11; Linux x86_64; rv:125.0) Gecko/20100101 Firefox/125.0"
[19/May/2024:17:57:04 +0000] <mydomain> <myip> <myuser> "PUT /webdav/browsersync/tabs_htpc.xbel.lock HTTP/2.0" 204 0 "-" "Mozilla/5.0 (X11; Linux x86_64; rv:125.0) Gecko/20100101 Firefox/125.0"
[19/May/2024:17:57:04 +0000] <mydomain> <myip> <myuser> "PUT /webdav/browsersync/tabs_htpc.xbel.lock HTTP/2.0" 204 0 "-" "Mozilla/5.0 (X11; Linux x86_64; rv:125.0) Gecko/20100101 Firefox/125.0"
[19/May/2024:17:57:37 +0000] <mydomain> <myip> <myuser> "PUT /webdav/browsersync/tabs_htpc.xbel.lock HTTP/2.0" 204 0 "-" "Mozilla/5.0 (X11; Linux x86_64; rv:125.0) Gecko/20100101 Firefox/125.0"
[19/May/2024:17:58:00 +0000] <mydomain> <myip> <myuser> "GET /webdav/browsersync/tabs_htpc.xbel.lock HTTP/2.0" 200 42 "-" "Mozilla/5.0 (X11; Linux x86_64; rv:125.0) Gecko/20100101 Firefox/125.0"
```
Client
```
"2024-05-19T17:46:53.259Z Setting lock: https://<mydomain>/webdav/browsersync/tabs_htpc.xbel.lock",
"2024-05-19T17:47:04.032Z Setting lock: https://<mydomain>/webdav/browsersync/tabs_htpc.xbel.lock",
"2024-05-19T17:47:04.036Z Setting lock: https://<mydomain>/webdav/browsersync/tabs_htpc.xbel.lock",
"2024-05-19T17:47:37.249Z Setting lock: https://<mydomain>/webdav/browsersync/tabs_htpc.xbel.lock",
"2024-05-19T17:48:53.259Z Setting lock: https://<mydomain>/webdav/browsersync/tabs_htpc.xbel.lock",
"2024-05-19T17:49:04.032Z Setting lock: https://<mydomain>/webdav/browsersync/tabs_htpc.xbel.lock",
"2024-05-19T17:49:04.035Z Setting lock: https://<mydomain>/webdav/browsersync/tabs_htpc.xbel.lock",
"2024-05-19T17:49:37.248Z Setting lock: https://<mydomain>/webdav/browsersync/tabs_htpc.xbel.lock",
"2024-05-19T17:50:53.260Z Setting lock: https://<mydomain>/webdav/browsersync/tabs_htpc.xbel.lock",
"2024-05-19T17:51:04.032Z Setting lock: https://<mydomain>/webdav/browsersync/tabs_htpc.xbel.lock",
"2024-05-19T17:51:04.037Z Setting lock: https://<mydomain>/webdav/browsersync/tabs_htpc.xbel.lock",
"2024-05-19T17:51:37.248Z Setting lock: https://<mydomain>/webdav/browsersync/tabs_htpc.xbel.lock",
"2024-05-19T17:52:53.261Z Setting lock: https://<mydomain>/webdav/browsersync/tabs_htpc.xbel.lock",
"2024-05-19T17:53:04.033Z Setting lock: https://<mydomain>/webdav/browsersync/tabs_htpc.xbel.lock",
"2024-05-19T17:53:04.037Z Setting lock: https://<mydomain>/webdav/browsersync/tabs_htpc.xbel.lock",
"2024-05-19T17:53:37.250Z Setting lock: https://<mydomain>/webdav/browsersync/tabs_htpc.xbel.lock",
"2024-05-19T17:54:53.261Z Setting lock: https://<mydomain>/webdav/browsersync/tabs_htpc.xbel.lock",
"2024-05-19T17:55:04.033Z Setting lock: https://<mydomain>/webdav/browsersync/tabs_htpc.xbel.lock",
"2024-05-19T17:55:04.038Z Setting lock: https://<mydomain>/webdav/browsersync/tabs_htpc.xbel.lock",
"2024-05-19T17:55:37.250Z Setting lock: https://<mydomain>/webdav/browsersync/tabs_htpc.xbel.lock",
"2024-05-19T17:56:53.261Z Setting lock: https://<mydomain>/webdav/browsersync/tabs_htpc.xbel.lock",
"2024-05-19T17:57:04.034Z Setting lock: https://<mydomain>/webdav/browsersync/tabs_htpc.xbel.lock",
"2024-05-19T17:57:04.038Z Setting lock: https://<mydomain>/webdav/browsersync/tabs_htpc.xbel.lock",
"2024-05-19T17:57:37.255Z Setting lock: https://<mydomain>/webdav/browsersync/tabs_htpc.xbel.lock",
"2024-05-19T17:58:00.094Z Starting sync process for account <myuser>@<mydomain>",
"2024-05-19T17:58:00.104Z onSyncStart: begin",
"2024-05-19T17:58:00.106Z https://<mydomain>/webdav/browsersync/tabs_htpc.xbel.lock",
"2024-05-19T17:58:00.181Z Resource is locked, trying again soon"
```

Firefox addon debug console is flooded with two types of messages:

    Uncaugt (in promise) Error: Could not establish connection. Receiving end does not exist.
    TypeError: n.default.action is undefined

If I filter down to "log" nothing is left.

**Comment by marcelklehr** [comment]
Nice catch! and this client is also on v5.1.3?

**Comment by G-Huber** [comment]
> Nice catch! and this client is also on v5.1.3?

Yes. All logs provided show clients with V5.1.3 - except the Android App which is 5.1.2 but shouldn't really matter here anyways.

**Comment by marcelklehr** [comment]
Damn, so my fix in v5.1.3 really didn't work. Can you try these builds here?

[floccus-build-v5.1.4-alpha.1-firefox.zip](https://github.com/floccusaddon/floccus/files/15370867/floccus-build-v5.1.4-alpha.1-firefox.zip)
[floccus-build-v5.1.4-alpha.1-chrome.zip](https://github.com/floccusaddon/floccus/files/15370868/floccus-build-v5.1.4-alpha.1-chrome.zip)

**Comment by andbenn** [comment]
> Also, do you have backups enabled for the current user in the bookmarks settings?

Floccus backing up bookmarks?  I don't see that in 5.1.3.  I do take browser bookmark backups weekly, so I'm relatively safe on that front.

**Comment by marcelklehr** [comment]
> Also, do you have backups enabled for the current user in the bookmarks settings?

Sorry, that comment belonged to another issue :D

**Comment by marcelklehr** [comment]
> My bookmarks has a .lock but my browsers seem to be running right over it.

Ah, that's a new issue due to my fix :see_no_evil:

**Comment by marcelklehr** [comment]
Can you try these builds, please? :pray: 

[floccus-build-v5.1.4-alpha.2-firefox.zip](https://github.com/floccusaddon/floccus/files/15373387/floccus-build-v5.1.4-alpha.2-firefox.zip)
[floccus-build-v5.1.4-alpha.2-chrome.zip](https://github.com/floccusaddon/floccus/files/15373388/floccus-build-v5.1.4-alpha.2-chrome.zip)

<details>
<summary>How to install alpha/beta versions</summary>


### Chrome (separate testing installation)
* Download the build here: https://github.com/marcelklehr/floccus/releases/tag/v4.2.0-beta.4
* Create a new folder
* Unzip the zip file there
* In chrome go to `chrome://extensions/`
* Enable developer mode
* Click `Load unpacked extension` and select the folder you created

### Chrome (direct upgrade; might not always work)
* Download the crx build here: https://github.com/marcelklehr/floccus/releases/tag/v4.2.0-beta.4
* In chrome go to `chrome://extensions/`
* Drop the `.crx` file into the page to install

### Firefox (separate testing installation; cannot upgrade)
* Download the zip build here: https://github.com/marcelklehr/floccus/releases/tag/v4.2.0-beta.4
* Unzip the zip file in a new folder
* In firefox go to `about:debugging`
* Click `Load tempoprary addon` and select the manifest file in the folder you created

</details>

**Comment by G-Huber** [comment]
I've been running my two clients on 5.1.4-alpha for the last ~13 hours.

lighttpd logs don't show a single entry where a GET was performed on a lock and answered with HTTP 200.
I'd interpret this as no lock getting stuck this time around. (For yesterday there are plenty of GET locks answered HTTP 200 by the server because a lock file already existed.)

I'll keep monitoring this throughout the week, but I won't be able keep both clients up all the time.

**Comment by marcelklehr** [comment]
Awesome, that is alpha.2 I hope? :)

**Comment by andbenn** [comment]
I may try the alpha 5.1.4-2 build soon.  Still on 5.1.3.  But I've got a different problem with my tab profile.  This is the second time this situation happened.

I've got a Brave tab profile unique to one browser.  I've got many browsers open, and each browser has many tabs.  Maybe 10 browsers, say 7 tabs average.  My autosync interval is 15 mins.  WebDav to fastmail.

I rebooted my Mac and things looked ok.  Stepped away and came back to the fan going.  Seems that the extension pulls the tab profile, and starts everything.  And hits a loop.  I have now probably about 90+ windows with 7 tabs in each, and a mac that sounds like it's going to take off.  And is not usable.  There are series of several browsers repeating over and over.  The previous case had a window that 4 tabs kept getting added to, over and over and over and over.

Rule of thumb: before rebooting or restarting one's browser, they may want to ensure the tab profile is good and syncing normally.  I think floccus was in "schedule" zone.

I managed to disable the extension which should stop additional windows from opening while my system tries to catch up.

**Comment by marcelklehr** [comment]
Yeah, I generally discourage people from using auto-sync with tabs profiles unless it's just to push changes.

**Comment by andbenn** [comment]
> Yeah, I generally discourage people from using auto-sync with tabs profiles unless it's just to push changes.

I will do that soon.  I did edit my comment (and forgot to save it) about potential tab name collisions causing issues.  And as the window name (in chromium browsers)

**Comment by marcelklehr** [comment]
I'm closing this for now, as v5.1.4 is released now, please shout if this does happen again!
